File: is_palindrome/is_palindrome.ts
export {}

function isAlphanumeric(char: string): boolean{
    const code = RegExp('^[a-zA-Z0-9]$');
    return code.test(char);
}

function isPalindrome(str: string): boolean{
    const length = str.length;
    let l: number = 0;
    let r: number = length - 1;

    while ( l < r ){
        while ( l < r && !isAlphanumeric(str[l])) l ++;
        while ( l < r && !isAlphanumeric(str[r])) r --;
        if(str[l].toLowerCase() !== str[r].toLocaleLowerCase()){
            return false;
        }
        l ++;
        r --;
    }
    return true;
}

const string: string = "A man, a plan, a canal: Panama";
console.log(isPalindrome(string));

File: is_palindrome/is_palindrome.py
def is_palindrome(s):
    length = len(s)
    l = 0
    r = length - 1
    while l < r:
        while l < r and not s[l].isalnum():
            l += 1
        while l < r and not s[r].isalnum():
            r -= 1
        if s[l].lower() != s[r].lower():
            return False
        l += 1
        r -= 1
    return True

str = "A man, a plan, a canal: Panama"
print("Is the string a palindrome?:", is_palindrome(str))